docker/generic/start_proxy.py

```diff
@@ -495,6 +495,16 @@ def make_argparser():
         connecting to Google service control. If it is `open`, the request will be allowed,
         otherwise, it will be rejected. Default is `open`.
         ''')
+    parser.add_argument(
+        '--enable_jwks_async_fetch',
+        action='store_true',
+        default=False,
+        help='''
+        Enable JWKS fetching for JWT authentication to be done before processing any requests.
+        If disabled, JWKS fetching is done when processing the requests. Recommended to
+        enable it. The default is disabled to use the old behaviors for not to break any
+        production systems.'''
```

Comment:
What do you think will break if we enable by default? I do not see much risk.

Reply:
My original concern is this case:  If the service config has a jwt_provider with unreachable remote_jwks_uri, but it is not used.    the old way is fine since it is not used, we will never try to fetch it.  In the new way,  we will try to fetch it,  but it fails
My concern is it will block listener to be active.   But I just checked the code,  we still activate listener after it fails.

So the only concern is delayed listener port binding since envoy has to fetch all jwks first before  binding the port. 
Some deployment may care.

Reply:
On the second though, if that is the only concern,  I will default this flag to be on